lib/Driver/DependencyGraph.cpp

```diff
@@ -302,13 +304,43 @@ void DependencyGraphImpl::markExternal(SmallVectorImpl<const void *> &visited,
       continue;
     assert(dependent.flags & DependencyFlags::IsCascading);
     visited.push_back(dependent.node);
-    markTransitive(visited, dependent.node);
+    markTransitive(visited, dependent.node, counters);
+  }
+}
+
+void
+DependencyGraphImpl::countMarking(CompilationCounters &counters,
+                                  DependencyMaskTy kinds, bool isCascading) {
+  if (isCascading) {
```

Comment:
This data is measuring "dependency edges actually traversed during a rebuild", right? Is that the right thing to be measuring? (As opposed to "total kinds of dependency edges", or "max fan-out from one changed file, regardless of edges taken to get there".)